docs/DeveloperGuide.md

```diff
@@ -1,25 +1,290 @@
+
+### List Function
+
+#### System Architecture of the List Function<br/>
+
+![](images/ListCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the list command.<br/>
+* UI: The User Interface of the app<br/>
+* ListCommand: The main logic command of the list function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the List Function<br/>
+
+![](images/ListCommand_umlDiagram.png)
+
+1. `ListCommand` gets the input from the User and gets the relevant user's timetable(s) from `Timetable`
+1. `ListCommand` then passes the user's timetable(s) to the `UI`
+1. `UI` prints all the classes in the timetable
+
+#### Sequence Design of List Function<br/>
+![](images/ListCommand_SequenceDiagram.png)<br/>
+
+### Find Function
+
+#### System Architecture of the Find Function<br/>
+
+![](images/FindCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the find command.<br/>
+* UI: The User Interface of the app<br/>
+* FindCommand: The main logic command of the find function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Find Function<br/>
+
+![](images/FindCommand_umlDiagram.png)
+
+1. `FindCommand` gets the input from the User and gets the timetable from `Timetable`
+1. `FindCommand` then searches through the timetable to find event that match the keyword entered by User
+1. `FindCommand` passes the matching events to `UI`
+1. `UI` prints the matching events
+
+#### Sequence Design of Find Function<br/>
+![](images/FindCommand_SequenceDiagram.png)<br/>
+
+### Compare Function<br/>
+#### System Architecture of the Compare Function<br/>
+![](images/CompareCommand_ClassDiagram.jpg) <br/>
+The System Architecture given above explains the high-level design of the <b>COMPARE</b> command.<br/>
+
+Listed below are some features we will be highlighting:<br/>
+
+* Ui: The User Interface of the app<br/>
+* CompareCommand: The main logic command of the <b>COMPARE</b> function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* User: User(s) of the program<br/>
+* UserList: An arraylist of User<br/>
+
+#### Design of the Compare Function<br/>
+![](images/CompareCommand_UML_Diagram.jpg)<br/>
+
+1) The <b>COMPARE</b> function gets the inputs from the User and retrieves the User's / Target User's<br/>
+timetables based on a specific day. It will then compare the timetables and return an arraylist
+of common timeslots.<br/>
+2)  It will then pass the arraylist onto Ui class to display the range of common <br/>
+timeslots the User and Target User have on a specific day of the week.<br/>
+
+
+#### Sequence Design of Compare Function<br/>
+![](images/CompareCommand_SequenceDesign.jpg) <br/>
```

Comment:
Arrows that indicate returning from the Ui class should be from the end of the activation bar, it should also reach the activation bar of the execute() in CompareCommand class. Also, try to use puml for diagrams as it would be neater compared to drawing it by hand.